### PR TITLE
allowed User assigned identity deployments

### LINKED ISSUE #1360: [Feature Request]: Support User Assigned Identity for Policy Assignment
### Description

Allow Policy assignments to leverage an existing user assigned identity as it may have the required permissions for the modify actions

### PR DESCRIPTION
**Pull request by ahmadabdalla**
# Description

Closes #1360 

Enables User assigned identity support for a policy assignment.

Note: This will not trigger the 'role assignment' resource creation object as it is assumed that a user identity will have the permissions required to modify resources. Hence the role assignment will only run if the policy is using a system assigned identity.

| Pipeline | Run |
| - | - |
| policy assignment |[Authorization: PolicyAssignments](https://github.com/Azure/ResourceModules/actions/workflows/ms.authorization.policyassignments.yml) |

# Type of Change

Please delete options that are not relevant.
- [X] New feature (non-breaking change which adds functionality)

# Checklist

- [ ] I'm sure there are no other open Pull Requests for the same update/change
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (readme)
- [ ] I did format my code

### CODE REVIEW

File: arm/Microsoft.Authorization/policyAssignments/.parameters/mg.parameters.json [hunk 1]
```diff
@@ -41,7 +41,7 @@
         },
         "notScopes": {
             "value": [
-                "/subscriptions/<<subscriptionId>>/resourceGroups/validation-rg"
+                "/subscriptions/<<subscriptionId>>/resourceGroups/<<resourceGroupName>>"
```
Comment (MrMCake) [inline arm/Microsoft.Authorization/policyAssignments/.parameters/mg.parameters.json:44]: Do we have this token? Haven't seen it used anywhere else

Comment (MrMCake) [inline arm/Microsoft.Authorization/policyAssignments/.parameters/mg.parameters.json:44]: Do we have this token? Haven't seen it used anywhere else

Comment (ahmadabdalla) [reply arm/Microsoft.Authorization/policyAssignments/.parameters/mg.parameters.json:44]: A couple of resources in the MS AuthZ namespace use it. We already support it and it's in our documentation. It replaces it with the resource group name from the YAML global variables. I can replace it back to the hard coded version if wanna keep using that?

Comment (eriqua) [reply arm/Microsoft.Authorization/policyAssignments/.parameters/mg.parameters.json:44]: I'd also suggest to have consistency on that and decide whether to use it everywhere or not.

Comment (ahmadabdalla) [reply arm/Microsoft.Authorization/policyAssignments/.parameters/mg.parameters.json:44]: I'll revert it and we can decide to roll it out for all pipelines in another issue instead as it isn't the reason for this PR.

Comment (ahmadabdalla) [reply arm/Microsoft.Authorization/policyAssignments/.parameters/mg.parameters.json]: removed token

**Review by MrMCake — APPROVED** [review]

**Pull request merged**